recipes/service.rb

```diff
@@ -23,6 +21,12 @@
 service 'dovecot' do
   service_name node['dovecot']['service']['name']
   supports Mash.new(node['dovecot']['service']['supports'])
-  provider node['dovecot']['service']['provider']
-  action [:enable, :start]
+  if node['platform'] == 'ubuntu' &&
+     Gem::Requirement.new(['>= 13.10', '<= 15.10'])
+                     .satisfied_by?(Gem::Version.new(node['platform_version']))
+    provider Chef::Provider::Service::Upstart
+  elsif node['platform'] == 'debian' && node['platform_version'].to_i < 9
+    provider Chef::Provider::Service::Debian
+  end
+  action %i[enable start]
```

Comment:
I see you moved the `node['dovecot']['service']['provider']` attribute logic from attributes file to the recipe. Is this a requirement of new Chef versions or just to avoid having code in attribute files? Previously we can override the provider, now we cannot.

Anyway, doing this change, will require to update the [Distribution Service Configuration](https://github.com/zuazo/dovecot-cookbook#distribution-service-configuration) section of the README. Probably other sections like [Supported Platforms](https://github.com/zuazo/dovecot-cookbook#supported-platforms) will also need to be updated.

Reply:
Oh, I did not see that comment. My fault.